client/dashboard/sites/settings-mcp/index.tsx

```diff
@@ -0,0 +1,290 @@
+import { queryClient } from '../../app/query-client';
+import PageLayout from '../../components/page-layout';
+import SettingsPageHeader from '../settings-page-header';
+import type { SiteMcpSettings } from '../../data/site-settings';
+
+export default function SettingsMcp( { siteSlug }: { siteSlug: string } ) {
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const { data: site } = useSuspenseQuery( siteBySlugQuery( siteSlug ) );
+	const { data: siteSettings, isLoading } = useQuery( siteSettingsQuery( site.ID ) );
+	const { data: isAutomattician } = useQuery( isAutomatticianQuery() );
+	const mutation = useMutation( siteSettingsMutation( site.ID ) );
+
+	// Gate access to Automatticians only
+	if ( ! isAutomattician ) {
+		throw notFound();
+	}
+
+	// Get abilities from the site settings data
+	const availableAbilities = Object.entries( siteSettings?.mcp_settings?.mcp_abilities || {} );
+	const hasAbilities = availableAbilities.length > 0;
+
+	const [ formData, setFormData ] = useState< SiteMcpSettings >( {
+		mcp_enabled: hasAbilities ? siteSettings?.mcp_settings?.mcp_enabled ?? true : false,
+		mcp_abilities: siteSettings?.mcp_settings?.mcp_abilities ?? {},
+	} );
+
+	// Update form data when siteSettings changes
+	useEffect( () => {
+		if ( siteSettings?.mcp_settings ) {
+			const newHasAbilities =
+				Object.keys( siteSettings.mcp_settings.mcp_abilities || {} ).length > 0;
+			setFormData( {
+				mcp_enabled: newHasAbilities ? siteSettings.mcp_settings.mcp_enabled : false,
+				mcp_abilities: siteSettings.mcp_settings.mcp_abilities,
+			} );
+		}
+	}, [ siteSettings?.mcp_settings ] );
+
+	// Auto-disable MCP if no abilities are selected
+	useEffect( () => {
+		const enabledAbilitiesCount = Object.values( formData.mcp_abilities ).filter(
+			( ability ) => ability.enabled
+		).length;
+
+		if ( formData.mcp_enabled && enabledAbilitiesCount === 0 ) {
+			setFormData( ( prev ) => ( {
+				...prev,
+				mcp_enabled: false,
+			} ) );
+		}
+	}, [ formData.mcp_abilities, formData.mcp_enabled ] );
+
+	const renderContent = () => {
+		// Show loading state while data is being fetched
+		if ( isLoading ) {
+			return (
+				<Card>
+					<CardBody>
+						<p>{ __( 'Loading MCP settings…' ) }</p>
+					</CardBody>
+				</Card>
+			);
+		}
+
+		const handleSubmit = ( e: React.FormEvent ) => {
+			e.preventDefault();
+
+			// Build the complete mcp_abilities object with all required properties
+			const mcp_abilities: Record<
+				string,
+				{ label: string; description: string; enabled: boolean }
+			> = {};
+
+			// Get the base abilities from site settings
+			const baseAbilities = siteSettings?.mcp_settings?.mcp_abilities || {};
+
+			// Merge with form data to get current enabled state
+			Object.entries( baseAbilities ).forEach( ( [ abilityId, ability ] ) => {
+				mcp_abilities[ abilityId ] = {
+					label: ability.label,
+					description: ability.description,
```

Comment:
It looks weird that we have to submit the label and description to the endpoint 😅 

Could we update the endpoint to accept something like?
```js
{
  [ abilityId ]: boolean
}
```